### PR TITLE
Revamped content and layout of project docs (README, CONTRIBUTING, etc)

### PR DESCRIPTION
**Pull request by smallsaucepan**
Felt like our project docs like README, CONTRIBUTING, etc were getting a bit messy. Have changed the layout and revised the content of many of them. Moved all but the publish to NPM non-negotiables (README, LICENSE and CHANGELOG) to the docs/ subdirectory.

As an example, split parts of README into GETTING_STARTED, and parts of CONTRIBUTING into PUBLISHING. Trimmed quite a bit of low level "type this command then this command" kind of thing.

Logic being some of these sections most contributors won't ever need to know, so better to keep what they will need as navigable as possible. And the step by step instruction sections are difficult to keep up to date or guarantee they'll work for everyone.

Updated the code of conduct from Contributor Covenant 1.4 to [2.1](https://www.contributor-covenant.org/version/2/1/code_of_conduct/). Left the wording as is e.g. didn't bother customising from "community" to "project" or "member" to "maintainer".

Also revamped issue and PR templates to ask questions I felt when doing my own PRs would be more meaningful to answer.

Possibly easier to browse the PR branch than look at the MD diffs below - https://github.com/smallsaucepan/turf/tree/proj-doc-updates

Please take a look. Might be going too far for some. If so, add a comment and maybe we can add some things back in.

Please provide the following when creating a PR:

- [x] Meaningful title, including the name of the package being modified.
- [x] Summary of the changes.
- [x] Heads up if this is a breaking change.
- [x] Any issues this [resolves](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests).
- [ ] Inclusion of your details in the `contributors` field of `package.json` - you've earned it! 👏
- [x] Confirmation you've read the steps for [preparing a pull request](https://github.com/Turfjs/turf/blob/master/docs/CONTRIBUTING.md#preparing-a-pull-request).

### CODE REVIEW

**Comment by mfedderly** [comment]
It is worth explicitly stating that this changes the code of conduct from [Contributor Covenant 1.4](https://www.contributor-covenant.org/version/1/4/code-of-conduct/) to [2.1](https://www.contributor-covenant.org/version/2/1/code_of_conduct/). I don't have a problem with this change, but it feels important to state explicitly here.

File: README.md [hunk 1]
```diff
@@ -15,104 +15,33 @@
...
-  },
-  properties: {}
-};
-```
+Other runtimes, such as Deno or Bun, are not officially supported. We would be very interested to hear your experiences though.
 
 ## Browser support
 
-Turf packages are compiled to target ES2017. However, the browser version of @turf/turf is transpiled to also include support for IE11. If you are using these packages and need to target IE11, please transpile the following packages as part of your build:
+Turf uses babel to transpile to a JavaScript version usable by most
```
Comment (mfedderly) [inline README.md:37]:
```suggestion
Turf uses Babel to transpile to a JavaScript version usable by most
```

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+JSDoc comments are found in the top-level index file for each module (for example `turf/packages/turf-area/index.ts`).  Look for the `@name`, `@params`, `@returns`, `@example` tags.
+
+## Code Style
+
+We have lots of tooling dedicated to ensuring consistent code. We use [Prettier](https://prettier.io/), [Typescript](https://www.typescriptlang.org/), and [ESLint](https://eslint.org/) to help us deliver quality code. These are checked by the build system and should be enforced at commit time by [Husky](https://typicode.github.io/husky/#/).
+
+Most modules are written in Typescript, while a few plain Javascript still linger. You can generally use modern Javascript when modifying Turf itself as all exported code is transpiled to (currently es2017).
+
+Code for consumption by browsers is transpiled by babel as described in the README.
```
Comment (mfedderly) [inline docs/CONTRIBUTING.md:129]:
```suggestion
Code for consumption by browsers is transpiled by Babel as described in the README.
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+The module README.md files and https://turfjs.org API docs are auto-generated from JSDoc comments in each modules' source file.
+
+As a contributor, all you need to do is keep the JSDoc comments up to date, and the rest is automated. For example if you change a top-level Turf functions signature (params, return type) or just want to improve the descriptiveness of the docs.
+
+JSDoc comments are found in the top-level index file for each module (for example `turf/packages/turf-area/index.ts`).  Look for the `@name`, `@params`, `@returns`, `@example` tags.
+
+## Code Style
+
+We have lots of tooling dedicated to ensuring consistent code. We use [Prettier](https://prettier.io/), [Typescript](https://www.typescriptlang.org/), and [ESLint](https://eslint.org/) to help us deliver quality code. These are checked by the build system and should be enforced at commit time by [Husky](https://typicode.github.io/husky/#/).
```
Comment (mfedderly) [inline docs/CONTRIBUTING.md:125]:
```suggestion
We have lots of tooling dedicated to ensuring consistent code. We use [Prettier](https://prettier.io/), [TypeScript](https://www.typescriptlang.org/), and [ESLint](https://eslint.org/) to help us deliver quality code. These are checked by the build system and should be enforced at commit time by [Husky](https://typicode.github.io/husky/#/).
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+Most modules are written in Typescript, while a few plain Javascript still linger. You can generally use modern Javascript when modifying Turf itself as all exported code is transpiled to (currently es2017).
```
Comment (mfedderly) [inline docs/CONTRIBUTING.md:127]:
```suggestion
Most modules are written in TypeScript, while a few plain Javascript still linger. You can generally use modern Javascript when modifying Turf itself as all exported code is transpiled to (currently es2017).
```

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        points.geojson
+```
+
+* `index.ts` - This file contains the definition of the function including [TypeScript](https://www.typescriptlang.org/) types, function implementation, the embedded [JSDocs](http://usejsdoc.org) documentation, and finally the single exported function that the module provides.
+* `bench.ts` - This file uses [Benchmark](https://benchmarkjs.com/) to establish benchmark performance tests for your function.
+* `test.ts` - This file includes your [tape](https://github.com/substack/tape)
+  unit tests. Tests can either be traditional code based tests, or tests that pull in data files and compare inputs to outputs (see below).
+* `package.json` - The [node](http://nodejs.org) metadata container file.
+  Modules imported in `index.ts` should be listed here under `dependencies`,
+  and modules used in `test.ts` or `bench.ts` should be listed under
```
Comment (mfedderly) [inline docs/CONTRIBUTING.md:69]: I was going to comment on just these two instances, but there's a lot of usages of "module" in this file that are probably more canonically called "packages". Might be worth switching around.

File: docs/CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official email address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[@morganherlocker](https://github.com/morganherlocker) or [@tmcw](https://github.com/tmcw).
```
Comment (mfedderly) [inline docs/CODE_OF_CONDUCT.md:64]: It looks like we maybe removed a few people here, is anyone interested in being added?

**Review by mfedderly — COMMENTED** [review]
This looks like a great refresh on the docs, I have a bunch of little nits.

Happy to approve but also happy to give others time to review as well.

File: docs/CONTRIBUTING.md (hunk 5 above)
Comment (smallsaucepan) [reply docs/CONTRIBUTING.md]: Updated to use consistent terminology.

**Review by twelch — APPROVED** [review]
I generally approve of these changes.  See my comment about potentially dropping the GETTING_STARTED in this repo and linking to the versioned one on the doc website.

File: README.md [hunk 7]
```diff
@@ -15,104 +15,33 @@
...
-  - > The Turf for Dart port is still in progress, the implementation status can be found in the [README](https://github.com/dartclub/turf_dart#components).
-- - -
-
-## Data in Turf
 
-Turf uses <a href='https://geojson.org/'>GeoJSON</a> for all geographic data. Turf expects the data to be standard <a href='https://en.wikipedia.org/wiki/World_Geodetic_System'>WGS84</a> longitude, latitude coordinates. Check out <a href='https://geojson.io/#id=gist:anonymous/844f013aae8354eb889c&map=12/38.8955/-77.0135'>geojson.io</a> for a tool to easily create this data.
+## Getting Started
 
-> **NOTE:** Turf expects data in (longitude, latitude) order per the GeoJSON standard.
+Read our [getting started guide](docs/GETTING_STARTED.md) to begin working with Turf.
```
[the remark this answers is not in the document]
Comment (smallsaucepan) [reply README.md]: Done. Thanks for the suggestion @twelch

**Comment by smallsaucepan** [comment]
Thanks @mfedderly and @twelch. Incorporated all your suggestions. Will merge.

**Pull request merged**